packages/components/src/presentation-components/properties/inputs/UniquePropertyValuesSelector.tsx

```diff
@@ -168,7 +172,12 @@ function useUniquePropertyValuesRuleset(descriptorRuleset?: Ruleset, field?: Fie
       return;
     }
 
-    const classInfos = getFieldClassInfos(field);
+    const classInfos = selectedClasses
+      ? selectedClasses?.map((classInfo) => ({
+          ...classInfo,
+          name: classInfo.name.replace(".", ":"),
```

Comment:
I would suggest updating `createSchemaClasses` to use `split(/[\.:]/)`.

Reply:
Thats a better solution, changed